lib/dgram.js

```diff
@@ -495,6 +495,19 @@ Socket.prototype.setMulticastLoopback = function(arg) {
 };
 
 
+Socket.prototype.setMulticastInterface = function(interfaceAddress) {
+  this._healthCheck();
+
+  if (typeof interfaceAddress !== 'string') {
```

Comment:
net.isIP() makes it look like no checks are necessary either in JS or the wrapper:

Its existing tests don't cover Scope, so I added some:
assert.equal(net.isIP('::%eth1'), 6);
assert.equal(net.isIP('::1%1'), 6);
These passed leading me to verify it is using a C implementation to get to libuv:
https://github.com/nodejs/node/blob/f8d3f6fbf4b0471677a0e4becf458a186fb2e4de/src/cares_wrap.cc#L34

yet, this doesn't need any wrapper:
https://github.com/nodejs/node/blob/863952ebadd4909a66bb9da7868bf75bbbe1462d/lib/net.js#L1581
and still passes:
assert.equal(net.isIP(), 0);

If net.isIP() is correct then no check on interfaceAddress in JS and removing the argument checks should be fine?